File: backend-spring-boot/src/main/java/no/vicx/backend/user/vm/RecaptchaResponseVm.java
package no.vicx.backend.user.vm;

import com.fasterxml.jackson.annotation.JsonProperty;

public record RecaptchaResponseVm(
        boolean success,
        @JsonProperty("challenge_ts")
        String challengeTimestamp,
        String hostname,
        @JsonProperty("error-codes")
        String[] errorCodes
) {
}

File: backend-spring-boot/src/main/kotlin/no/vicx/backend/user/vm/RecaptchaResponseVm.kt
package no.vicx.backend.user.vm

import com.fasterxml.jackson.annotation.JsonProperty


// See https://developers.google.com/recaptcha/docs/verify
data class RecaptchaResponseVm(
    val success: Boolean = false,
    @JsonProperty("challenge_ts") val challengeTimestamp: String? = null,
    val hostname: String? = null,
    @JsonProperty("error-codes") val errorCodes: List<String>? = null
)
